content/en/opentelemetry/interoperability/instrumentation_libraries.md

```diff
@@ -77,12 +65,39 @@ To use an OpenTelemetry instrumentation with the Datadog Java SDK:
 2. Copy the OpenTelemetry extension JAR containing the instrumentation to the same container as the application.
 3. Set the `otel.javaagent.extensions` system property or the `OTEL_JAVAAGENT_EXTENSIONS` environment variable to the extension JAR path.
 
+Here’s a step-by-step example using R2DBC in Java to illustrate how you can drop OpenTelemetry instrumentation into your service and begin sending data to Datadog, ensuring you capture all the missing spans. R2DBC is the reactive version of JDBC, commonly used for reactive database interactions in Java applications. Datadog’s Java agent does not provide out-of-the-box instrumentation for R2DBC queries. 
+
+Download the OpenTelemetry R2DBC agent and run your Spring Boot application with both the Datadog Java agent and the OpenTelemetry R2DBC agent.
+
+
+```sh
+git clone https://github.com/eugenp/tutorials
+cd tutorials/spring-reactive-modules/spring-reactive-data
+
+curl -Lo dd-java-agent.jar 'https://dtdg.co/latest-java-tracer'
+
+curl -Lo opentelemetry-javaagent-r2dbc.jar \
+  'https://repo1.maven.org/maven2/io/opentelemetry/javaagent/instrumentation/opentelemetry-javaagent-r2dbc-1.0/2.5.0-alpha/opentelemetry-javaagent-r2dbc-1.0-2.5.0-alpha.jar'
+
+mvn spring-boot:run -Dstart-class=com.baeldung.pagination.PaginationApplication \
+  -Dspring-boot.run.jvmArguments='-javaagent:dd-java-agent.jar -Ddd.trace.otel.enabled=true -Dotel.javaagent.extensions=opentelemetry-javaagent-r2dbc.jar -Ddd.trace.split-by-tags=db.name,db.sql.table -Ddd.trace.debug=true'
+```
+Open `http://127.0.0.1:8080/products` to exercise the product query. With this setup, you’re now using OpenTelemetry’s instrumentation to ensure full observability for R2DBC queries.
+
+<div class="alert alert-warning">
+Additionally, note that the `-Ddd.trace.split-by-tags=db.name,db.sql.table` setting gives R2DBC spans a service name based on the table. If you don't want R2DBC spans to appear under a separate service then that setting can be dropped.
+</div>
+
+<div class="alert alert-warning">
```

Comment:
Suggest moving this alert to ## Compatibility requirements.

Reply:
Done!